cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -170,6 +170,10 @@ func (m *azureCache) regenerate() error {
 		if err != nil {
 			return err
 		}
+		if instances == nil {
+			klog.V(4).Infof("regenerate: no nodes found for node group %s", ng.Id())
+			return nil
```

Comment:
Should we return here?
It looks like this loop is going to each group. Not sure if we want to halt the entire loop/method when one group is empty. Maybe `continue` is more appropriate?

Reply:
yeah agreed that makes more sense

Reply:
Yea, I think we must not return here. It could effectively disable autoscaling on other node groups. Let's be extra careful on the decision to return early.
Although, I feel like unit tests should have caught it. Do you think it is useful to add one in this case?